### PR TITLE
Update Driver API

### PR DESCRIPTION
**Pull request by fleupold**
# Description
The driver API is not up to date.

# Changes

- [x] Add reveal endpoint
- [x] Update all existing fields
- [x] Add feePolicy fields
- [x] Flatten and tag feePolicy enum in-line with remaining enums (e.g. Liquidity)

## How to test
Pasted the text in to see it render correctly

## Related Issues

Part of #2182

### CODE REVIEW

File: crates/driver/openapi.yml [hunk 1]
```diff
@@ -151,49 +170,67 @@ components:
             The reference price denominated in native token (i.e. 1e18 represents a token that
             trades one to one with the native token). These prices are used for solution competition
             for computing surplus and converting fees to native token.
-        available_balance:
-          $ref: "#/components/schemas/TokenAmount"
         trusted:
           type: boolean
```
Comment (MartinquaXD) [inline crates/driver/openapi.yml:174]: While we are at it we should probably document what exactly `trusted` means.

File: crates/driver/openapi.yml [hunk 2]
```diff
@@ -279,94 +337,110 @@ components:
...
-            Signature confirming that the Solver promised to have this solution for this auction.
+              This id is used to identify the solution when executing it.
+            type: integer
+          score:
+            description: |
+              The objective value of the solution.
+            type: number
+          submissionAddress:
+            description: |
+              The address that should be used to submit the solution.
```
Comment (MartinquaXD) [inline crates/driver/openapi.yml:363]: Let's make it more clear that the protocol expects the information to be correct.
```suggestion
              The address that will be used to submit the solution.
```

File: crates/driver/openapi.yml [hunk 3]
```diff
@@ -279,94 +337,110 @@ components:
...
+            $ref: "#/components/schemas/Address"
+          orders:
+            description: executed orders
+            type: array
```
Comment (MartinquaXD) [inline crates/driver/openapi.yml:367]: Don't these get serialized as an object with `key == uid` and `value == executedAmount`?

File: crates/driver/openapi.yml [hunk 4]
```diff
@@ -279,94 +337,110 @@ components:
...
-                  $ref: "#/components/schemas/TokenAmount"
-        calldata:
-          description: hex encoded
+            This is the hash that should be used to submit the solution.
           type: string
-          example: "0x01"
-        signature:
-          description: |
-            Signature confirming that the Solver promised to execute this auction.
+          example: "0x2fa8a928efdd11fcf09e196dfa6e3ac4884967a9409ecc1b854ed2eabe4839e2"
```
Comment (MartinquaXD) [inline crates/driver/openapi.yml:408]: I think `calldata` is an object like:
```json
{
    "internalized": "0xabcd",
     "uninternalized": "0xabcdefgh",
}
```

Comment (fleupold) [reply crates/driver/openapi.yml:408]: This was moved and is now explaining tx_hash which is just that.

**Review by MartinquaXD — APPROVED** [review]

**Review by squadgazzz — APPROVED** [review]

**Pull request merged**